### PR TITLE
Closes #393 Extra header line removed

### LINKED ISSUE #393: Bug: Incorrect `assert_integer_scalar()` test
We currently have a short section for unit testing the `assert_integer_scalar()` function. But in that section, we're accidentally testing `assert_numeric_vector()` instead 😱 

After the {cli} messaging updates are made, we should clean this up and probably add another test or two.

```r
# assert_integer_scalar ----
## Test 38: error if `arg` is not an integer scalar ----
test_that("assert_integer_scalar Test 38: error if `arg` is not an integer scalar", {
  example_fun <- function(arg) {
    assert_numeric_vector(arg)
  }

  arg <- c()

  expect_error(example_fun(TRUE))
  expect_error(example_fun(arg))
  expect_error(example_fun("1.5"))
})
```

### PR DESCRIPTION
**Pull request by ddsjoberg**
This turns out to just be an extra line added by accident. The checks for `# assert_integer_scalar ----` appear above this chunk.

Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/dev/CONTRIBUTING.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [x] Place Closes #393 into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` under the header `# admiral (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiraldev/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Review by bms63 — COMMENTED** [review]
Something seems amiss here?

**Comment by ddsjoberg** [comment]
> Something seems amiss here?

There were two sections for `# assert_integer_scalar ----`. Just deleting the second one that was out of place. Is there something else amiss @bms63 ?

File: tests/testthat/test-assertions.R [hunk 1]
```diff
@@ -687,7 +687,6 @@ test_that("assert_numeric_vector Test 39: no error if `arg` is NULL and optional
   )
 })
 
-# assert_integer_scalar ----
 ## Test 40: error if `arg` is not an integer scalar ----
 test_that("assert_integer_scalar Test 40: error if `arg` is not an integer scalar", {
```
Comment (bms63) [inline tests/testthat/test-assertions.R:691]: This seems inccorrect - the integer and numeric stuff

Comment (ddsjoberg) [reply tests/testthat/test-assertions.R]: ah, yes, the text needed to updated too
image

**Review by bundfussr — COMMENTED** [review]
I think there are more issues. For example `assert_character_scalar()` is tested in tests 10 and 11 and 17-19. Similar for `assert_data_frame()`. Could we clean this up?

And there are some `expect_error()` calls which accept any error, e.g.,
```
expect_error(example_fun2(2))
```
in test 10. I think we should always specify the `class` argument for `expect_error()`.

**Comment by ddsjoberg** [comment]
@bundfussr do you want to add those cleanups in this PR or another? If this PR, please feel free to push the updates here.

**Comment by bundfussr** [comment]
> @bundfussr do you want to add those cleanups in this PR or another? If this PR, please feel free to push the updates here.

I can add them to this PR.

**Comment by bundfussr** [comment]
@bms63 , @ddsjoberg , I have cleaned up `test-assertions.R`. Now there shouldn't be more than one section per function and all `expect_error()` calls should use the `class` argument.

Could you double-check that I didn't miss anything or mess up something?

**Review by bms63 — APPROVED** [review]
Thanks all!!

**Comment by ddsjoberg** [comment]
> @ddsjoberg wasn't there a tracking issue for all the error messaging updates? I can't seem to find it

@bms63 
There wasn't a master issue for all the updates. It was just a single issue listing the first round updates (https://github.com/pharmaverse/admiraldev/issues/367)

**Pull request merged**